lib/active_merchant/billing/gateways/alelo.rb

```diff
@@ -110,8 +110,14 @@ def fetch_access_token
           'Content-Type' => 'application/x-www-form-urlencoded'
         }
 
-        parsed = parse(ssl_post(url('captura-oauth-provider/oauth/token'), post_data(params), headers))
-        Response.new(true, parsed[:access_token], parsed)
+        begin
+          raw_response = ssl_post(url('captura-oauth-provider/oauth/token'), post_data(params), headers)
+        rescue ResponseError => e
+          raise OAuthResponseError.new(e)
+        else
+          parsed = parse(raw_response)
+          Response.new(true, parsed[:access_token], parsed)
```

Comment:
Should we check that `parsed[:access_token` is defined and if it isn't raise an `OAuthResponseError`?